tests/test_app.py

```diff
@@ -766,12 +907,88 @@ async def test_file_decompression(contents):
                 assert not os.path.exists(f1)
                 assert os.path.exists(os.path.join(d, name))
                 resp = await cli.patch(
-                    '/decompress/'+os.path.join(dirname, name),
-                    headers={'Authorization': ''})
+                    "/decompress/" + os.path.join(dirname, name),
+                    headers={"Authorization": ""},
+                )
                 assert resp.status == 200
                 text = await resp.text()
-                assert 'succesfully decompressed' in text
+                assert "succesfully decompressed" in text
                 assert name in text
                 # check to see if we got back what we started with for all files and directories
                 assert os.path.exists(d)
                 assert os.path.exists(f1)
+
+
+async def test_importer_mappings():
+    """
+    This tests calling with simple good cases, and some expected bad cases
+    :return:
+    """
+    username = "testuser"
+
+    # Normal case, no match
+    data = {"file_list": ["file1.txt"]}
+    async with AppClient(config, username) as cli:
+        resp = await cli.post(
+            "importer_mappings/", headers={"Authorization": ""}, data=data
+        )
+        assert resp.status == 200
+        text = await resp.json()
+        assert "mappings" in text
+        mappings = text["mappings"]
+        assert mappings[0] is None
+
+    # Normal case, one match
+    data = {"file_list": ["file1.txt", "file.zip"]}
+    async with AppClient(config, username) as cli:
+        resp = await cli.post(
+            "importer_mappings/", headers={"Authorization": ""}, data=data
+        )
+        assert resp.status == 200
+        text = await resp.json()
+        assert "mappings" in text
+        mappings = text["mappings"]
+        assert mappings[0] is None
+        # As we update the app mappings this test may need to be changed
+        # Or we need to reload json file itself
+
+        unzip_mapping = AutoDetectUtils._MAPPINGS["apps"]["Decompress/Unpack"]["id"]
+        assert mappings[1][0] == [unzip_mapping, 1]
+
+    # No files passed in
+    data = {}
+    async with AppClient(config, username) as cli:
+        resp = await cli.post(
+            "importer_mappings/", headers={"Authorization": ""}, data=data
+        )
+        assert resp.status == 200
```

Comment:
400 for empty input?

Reply:
@ialrmedaliendalien for empty input you get back the list of apps

Reply:
So there is no way to get any response other than 200, success, if you submit a query with the appropriate auth info?

Reply:
No auth is required to use this endpoint. I cant think of a way to get one other than 200 right now, since it always returns a list of available apps